python/mlcroissant/mlcroissant/_src/structure_graph/base_node.py

```diff
@@ -38,12 +39,14 @@ class Node(abc.ABC):
     """
 
     ctx: Context = dataclasses.field(default_factory=Context)
-    id: str = dataclasses.field(default_factory=generate_uuid)
+    id: str | Uuid = dataclasses.field(default_factory=generate_uuid)
```

Comment:
`id: Uuid`

We should try to avoid compound types as much as possible

Reply:
I agree, but this is actually handy internally -- e.g. in metadata.py:390 we would need to first convert the jsonld to a string, then use it as input for id. Now it is more compact...